src/sql/workbench/contrib/notebook/browser/outputs/gridOutput.component.ts

```diff
@@ -144,8 +146,43 @@ export class GridOutputComponent extends AngularDisposable implements IMimeCompo
 			this._layoutCalledOnce = true;
 		}
 	}
+	reorderGridData(source: IDataResource) {
+		let rowIndex = 0;
+		// Get schema list of grid
+		const columnNames: string[] = source.schema.fields.map(field => field.name);
+		if (source.data.length > 0) {
+			// Check to see if data source is ordered properly based on schema
```

Comment:
I really think this deserves more of an explanation of in what scenarios the data is unordered and why

Reply:
Sure, the main scenario being papermill execution with KQL as the issue states. But this will also make it robust when there are other executed grids via a headless client through papermill. Ill be sure to document it.